### PR TITLE
Refactor: is market listed for core pool logic

### PR DESCRIPTION
**Pull request by Debugger022**
- Refactored the logic for the isAssetListed in any market of the core pool.
- Refactored the `setPoolsAssetsDirectTransfer` method

### CODE REVIEW

File: contracts/TokenConverter/RiskFundConverter.sol [hunk 1]
```diff
@@ -72,22 +73,31 @@ contract RiskFundConverter is AbstractTokenConverter {
...
+    /// @custom:error InvalidArguments thrown when comptrollers array length is not equal to assets array length
     /// @custom:access Restricted by ACM
-    function setPoolsAssetsDirectTransfer(address[] calldata comptrollers, address[][] calldata assets) external {
-        _checkAccessAllowed("setPoolsAssetsDirectTransfer(address[], address[][])");
+    function setPoolsAssetsDirectTransfer(
+        address[] calldata comptrollers,
+        address[][] calldata assets,
+        bool[][] calldata values
+    ) external {
+        _checkAccessAllowed("setPoolsAssetsDirectTransfer(address[], address[][], bool[][])");
```
Comment (chechu) [inline contracts/TokenConverter/RiskFundConverter.sol:83]:
```suggestion
        _checkAccessAllowed("setPoolsAssetsDirectTransfer(address[],address[][],bool[][])");
```

File: contracts/TokenConverter/RiskFundConverter.sol [hunk 2]
```diff
@@ -72,22 +73,31 @@ contract RiskFundConverter is AbstractTokenConverter {
...
+            bool[] memory assetsValues = values[i];
+
+            if (poolAssets.length != assetsValues.length) {
+                revert InvalidArguments();
+            }
+
             for (uint256 j; j < poolAssets.length; ++j) {
-                poolsAssetsDirectTransfer[comptrollers[i]][poolAssets[j]] = true;
+                poolsAssetsDirectTransfer[comptrollers[i]][poolAssets[j]] = assetsValues[j];
                 emit PoolAssetsDirectTransferUpdated(comptrollers[i], poolAssets[j]);
```
Comment (chechu) [inline contracts/TokenConverter/RiskFundConverter.sol:101]: I would add `assetsValues[j]` into the event

File: contracts/TokenConverter/RiskFundConverter.sol (hunk 1 above)
Comment (Debugger022) [reply contracts/TokenConverter/RiskFundConverter.sol]: Fixed, https://github.com/VenusProtocol/protocol-reserve/pull/13/commits/3d64345e4d1d8ccf397b2c4439b5cfc2d34968bc

File: contracts/TokenConverter/RiskFundConverter.sol (hunk 2 above)
Comment (Debugger022) [reply contracts/TokenConverter/RiskFundConverter.sol]: Added, https://github.com/VenusProtocol/protocol-reserve/pull/13/commits/3d64345e4d1d8ccf397b2c4439b5cfc2d34968bc

**Pull request merged**